pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -2469,7 +2470,7 @@ private String receiveCommandStatus() throws IOException {
   private void interpretCommandStatus(String status, ResultHandler handler) {
     long oid = 0;
     long count = 0;
-    Matcher matcher = COMMAND_COMPLETE_PATTERN.matcher(status);
+    Matcher matcher = COMMAND_COMPLETE_MATCHER.reset(status);
```

Comment:
This would not appear to be thread safe.
A concurrent thread could call reset with a different status, affecting this thread’s call to matches or group methods.

Reply:
Matcher is not thread safe

Reply:
That is true, updated